### DISCUSSION TITLE
How can I include work items from other repos that are used by the YAML pipeline?

### DISCUSSION THREAD

**Discussion by magoicochea** [rfennell/AzurePipelines#1291] [Q&A]
Hello

At the moment I have three repos on my project

- **Infrastructure**: Stores the Infrastructure as Code (ARM Templates) and Deployment Scripts.
- **Backend**: Has the C# code of my function apps.
- **UI**: Stores my React Website

The app is deployed from an azure YAML Multi-Stage pipeline that is stored on the Infrastructure repo. This pipeline has the other two repos declared as resources. However, when the task is run it only gets the work items related to the repo where the YAML file is stored (Infrastructure in this case).

I do see some logs that make it sound as if there is a configuration missing on Azure DevOps:

> Cannot get extra commit or work item details of the Repository artifact backend null
> Cannot get extra commit or work item details of the Repository artifact UI null

Is there any permission or parameter that I am missing?

If I click on the "Related" link on the build I can see the work items that are not being included

![image](https://user-images.githubusercontent.com/260811/165284599-ddd378e4-b491-485f-91a7-9b8f40d55066.png)

EDIT:

I forgot to add the current task on the YAML pipeline

```
- task: XplatGenerateReleaseNotes@3
  displayName: 'Generate Release Notes'
  inputs:
    outputfile: '$(Pipeline.Workspace)\${{ parameters.OutputFile }}'
    templateLocation: InLine
    inlinetemplate: |
     # Release note for {{buildDetails.definition.name}} deployment
     * Build name : [{{buildDetails.buildNumber}}]({{buildDetails._links.web.href}})
     * Compared release name : [{{compareBuildDetails.buildNumber}}]({{compareBuildDetails._links.web.href}})
     * Primary artifact name : {{buildDetails.definition.name}}               
     ## Work items included ({{workItems.length}})
     {{#forEach this.workItems}}
     
     1. #{{this.id}}
     {{#forEach this.relations}}
     {{#if (contains this.attributes.name 'Parent')}}
     {{#with (lookup_a_work_item ../../relatedWorkItems  this.url)}}
  	 Parent : #{{this.id}}
     {{/with}}
     {{/if}}
     {{/forEach}} 
     {{/forEach}}
    checkStage: true
    overrideStageName: 'Prod'
    sortWi: true
    showOnlyPrimary: false
    dumpPayloadToConsole: true
    getParentsAndChildren: true
```

**Comment by rfennell** [comment]
My task uses the basic Microsoft API calls to find the commits/wi between builds, so if that is not returning the required items there is not much I can do.

But, can you upload a full copy of the log, with the debug level of logging (a variable system.debug=true, or check the diagnostics box if you manually queue the build) and I will have a detailed look

**Reply by magoicochea** [reply]
Hi

I managed to find a workaround, which was to include the YAML task after each of the builds that were calling the release pipeline and then using each of the resulting files to generate a bigger report. It was a bit of work, but it did the trick.

I also did what you ask me and here is the log of what I got when I added the system.debug variable

[buildLogs.log](https://github.com/rfennell/AzurePipelines/files/8564569/buildLogs.log)

**Reply by rfennell** [reply]
Actually the way you have done it is the way I usually do the tasks too.

Looking at you log, the key line is

> 2022-04-26T14:43:45.6016257Z ##[debug]Getting the details of the changes between the first build 56913 and the current build 64628

You will only get commits/WI associated between these two builds on the primary repo. So again you workaround technique is the best option

**Reply by magoicochea** [reply]
Ok, thanks for you help!
